src/basilisp/lang/compiler/analyzer.py

```diff
@@ -996,6 +1019,13 @@ def _def_ast(  # pylint: disable=too-many-locals,too-many-statements
     # its meta may contain values which fail to compile.
     bare_name = sym.symbol(name.name)
 
+    # Warn if this symbol is potentially being redefined (if the Var was
+    # previously bound)
+    if __should_warn_on_redef(current_ns, bare_name, def_meta):
+        logger.warning(
+            f"redefining Var '{bare_name}' in namespace {current_ns}:{def_loc[0]}"
+        )
+
```

Comment:
It never made sense to do this in the generator since the problem exists on the Basilisp side, not the Python side.

But to make matters worse, by checking this in the generator we were basically guaranteeing this feature would never work as intended. The analyzer always performs a `Var.intern_unbound` which will overwrite any Var metadata, so we'd lose any `^:redef` metadata by the time we were checking this condition in the generator.